runtime/autoload/man.vim

```diff
@@ -2,15 +2,24 @@
 
 let s:man_find_arg = "-w"
 
-" TODO(nhooyr) Completion may work on SunOS; I'm not sure if `man -l` displays
-" the list of searched directories.
-try
-  if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~# '^5'
-    let s:man_find_arg = '-l'
-  endif
-catch /E145:/
-  " Ignore the error in restricted mode
-endtry
+function s:init() abort
+  " TODO(nhooyr) Completion may work on SunOS; I'm not sure if `man -l` displays
+  " the list of searched directories.
+  try
+    if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~# '^5'
+      let s:man_find_arg = '-l'
```

Comment:
Looks like '-l' does something different on these platforms but I have no way to verify that so I kept it that way to avoid breakage.